lib/kafka_ex/server_kayrock/state.ex

```diff
@@ -0,0 +1,148 @@
+defmodule KafkaEx.ServerKayrock.State do
+  @moduledoc false
+
+  # state struct for ServerKayrock
+
+  alias KafkaEx.New.ClusterMetadata
+
+  defstruct(
+    bootstrap_uris: [],
+    cluster_metadata: %ClusterMetadata{},
+    correlation_id: 0,
+    consumer_group_for_auto_commit: nil,
+    metadata_update_interval: nil,
+    consumer_group_update_interval: nil,
+    worker_name: KafkaEx.Server,
+    ssl_options: [],
+    use_ssl: false,
+    api_versions: %{},
+    allow_auto_topic_creation: true
```

Comment:
This seems like it should be somewhere else, since it's a broker setting. We don't hold other broker settings that are even more important, like the maximum message size.

Reply:
This is a field in the metadata request as of v4.  It's `AND`ed with the cluster setting IIUC.  I think long-term I'd be completely OK with disabling this as a setting, but a lot of our existing tests depend on auto topic creation (which is not great).  So it's either allow it now and plan to remove it once we drop compatibility, or rework a ton of the compatibility tests.  I think I'm kind of OK with either approach but leaving it for now is a lot less work.